### DISCUSSION TITLE
Handling strictly "two-dose" vaccinated columns and incomplete data from request query

### DISCUSSION THREAD

**Discussion by smcclure17** [covid-projections/can-scrapers#199] [Scraper help] [answered]
[South Dakota](https://doh.sd.gov/COVID/Dashboard.aspx) returns data on people vaccinated with two doses, which does not include J&J/single-shot vaccines. As such, this doesn't match our `total_vaccine_completed` schema. They do have a "Total People Vaccinated" column that matches the `total_vaccine_initiated` definition. I'm able to collect and use this data without issue. 

However, they do _also_ surface data on single-shot vaccinations, so it is possible to get the data we want. Using a different query I've been able to get _some_ of this data. The query returns the # of and the type of (first-shot or second-shot) doses for each manufacturer, which could then be summed/combined to match our definitions. 

The query below (using the `MicrosoftBIDashboard` class) returns the correct data for the first 13 counties, but nothing for the remaining. I'm unsure as to why--Is there a default limit to the amount of data that can be returned? 

I can provide the output/response of the request if it would be helpful.

```
        body["queries"] = [
            {
                "Query": {
                    "Commands": [
                        {
                            "SemanticQueryDataShapeCommand": {
                                "Query": {
                                    "Version": 2,
                                    "From": self.construct_from(
                                        [
                                            # From
                                            ('c', 'County', 0),
                                            ('v', 'Vaccines', 0),
                                            ('m', ' Measures', 0)                                        
                                        ]
                                    ),
                                    "Select": self.construct_select(
                                        [
                                            # Selects
                                            ('c', 'County', 'county'),
                                            ('v', 'Manufacturer - Dose # (spelled out)', 'doses'),
                                        ],
                                        [],
                                        [
                                            # Measures
                                            ('m', 'Number of Recipients', 'recipients')
                                        ],
                                    ),
                                }
                            }
                        }
                    ]
                },
                "QueryId": "",
                "ApplicationContext": self.construct_application_context(
                    ds_id, report_id
                ),
            }
        ]
```

**My questions/problems:**
- Is it worth continuing to try to collect the manufacturer-specific data, or should we use the second-shot data that excludes the J&J doses?
- If we do want the more specific data, need to determine why not all of the counties are being included/returned in the query.
- Does the priority of this data change with the new `series complete` [CDC data](https://covid.cdc.gov/covid-data-tracker/COVIDData/getAjaxData?id=vaccination_county_condensed_data)?

This might be pretty unclear so I can clarify about anything mentioned

**Comment by sglyon** [comment, marked as answer]
Hey @smcclure17 I think my preference would be to use the "easy to get" resource for `total_vaccine_initiated` and then the CDC endpoint for `total_vaccine_completed`

Happy to hear thoughts from @ghop02  or @mikelehen though. Are we ok using the CDC as our only source for vaccines completed?

**Reply by mikelehen** [reply]
Starting with just `total_vaccine_initiated` sounds reasonable.  It'd be nice to get completed data too, but if there are blockers / obstacles, punting on that for now feels okay.  We could start pulling in the CDC data (though we haven't validated it yet and there may be e.g. reporting delays that make it undesirable to mix local / federal data sources), else we can revisit the issues here later.

**Reply by sglyon** [reply]
Sounds great -- looks like we can move forwards with just "at least one dose" @smcclure17 !

**Reply by smcclure17** [reply]
sounds good, thanks guys
